clients/client-java/src/test/java/org/apache/gravitino/client/integration/test/authorization/AccessControlIT.java

```diff
@@ -314,6 +316,57 @@ void testManageGroupPermissions() {
     metalake.deleteRole(roleName);
   }
 
+  @Test
+  void testManageRolePermissions() {
+    String roleName = "role#123";
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put("k1", "v1");
+    metalake.createRole(roleName, properties, Lists.newArrayList());
+    MetadataObject metadataObject =
+        MetadataObjects.of(null, metalakeName, MetadataObject.Type.METALAKE);
+
+    // grant a privilege
+    Role role =
+        metalake.grantPrivilegesToRole(
+            roleName, metadataObject, Lists.newArrayList(Privileges.CreateCatalog.allow()));
+    Assertions.assertEquals(1, role.securableObjects().size());
+
+    // repeat to grant a privilege
+    role =
+        metalake.grantPrivilegesToRole(
+            roleName, metadataObject, Lists.newArrayList(Privileges.CreateCatalog.allow()));
+    Assertions.assertEquals(1, role.securableObjects().size());
+
+    // grant a not-existing role
+    Assertions.assertThrows(
+        NoSuchRoleException.class,
+        () ->
+            metalake.grantPrivilegesToRole(
+                "not-exist", metadataObject, Lists.newArrayList(Privileges.CreateCatalog.allow())));
+
+    // revoke a privilege
+    role =
+        metalake.revokePrivilegesFromRole(
+            roleName, metadataObject, Lists.newArrayList(Privileges.CreateCatalog.allow()));
+    Assertions.assertTrue(role.securableObjects().isEmpty());
+
+    // repeat to revoke a privilege
+    role =
+        metalake.revokePrivilegesFromRole(
+            roleName, metadataObject, Lists.newArrayList(Privileges.CreateCatalog.allow()));
+    Assertions.assertTrue(role.securableObjects().isEmpty());
+
+    // revoke a not-existing role
+    Assertions.assertThrows(
+        NoSuchRoleException.class,
+        () ->
+            metalake.revokePrivilegesFromRole(
+                "not-exist", metadataObject, Lists.newArrayList(Privileges.CreateCatalog.allow())));
```

Comment:
Should you add the tests about binding wrong privileges to the metadata object?

Reply:
You'd better add more bad cases to test the robustness of your logic.

Reply:
Added.